src/test/java/com/google/api/codegen/testdata/csharp_gapic_resourcenames_library.baseline

```diff
@@ -0,0 +1,719 @@
+    {
+        public static UnknownResourceName Parse(string value) => new UnknownResourceName(value);
+
+        public static bool TryParse(string value, out UnknownResourceName result)
+        {
+            result = new UnknownResourceName(value);
+            return true;
+        }
+
+        public UnknownResourceName(string value)
+        {
+            Value = GaxPreconditions.CheckNotNull(value, nameof(value));
+        }
+
+        public string Value { get; }
+
+        public override string ToString() => Value;
+    }
+
+    public sealed class ResourceNameList<TResource> : IList<TResource> where TResource : IResourceName // TODO: This will be in GAX
+    {
+        public ResourceNameList(IList<string> underlyingList, Func<TResource, string> nameToString, Func<string, TResource> stringToName)
+        {
+            _underlyingList = GaxPreconditions.CheckNotNull(underlyingList, nameof(underlyingList));
+            _nameToString = GaxPreconditions.CheckNotNull(nameToString, nameof(nameToString));
+            _stringToName = GaxPreconditions.CheckNotNull(stringToName, nameof(stringToName));
+        }
+
+        private IList<string> _underlyingList;
+        private Func<TResource, string> _nameToString;
+        private Func<string, TResource> _stringToName;
+
+        public int Count => _underlyingList.Count;
+        public bool IsReadOnly => _underlyingList.IsReadOnly;
+        public TResource this[int i]
+        {
+            get { return _stringToName(_underlyingList[i]); }
+            set { _underlyingList[i] = _nameToString(value); }
+        }
+        public void Add(TResource value) => _underlyingList.Add(_nameToString(value));
+        public void Clear() => _underlyingList.Clear();
+        public bool Contains(TResource value) => _underlyingList.Contains(_nameToString(value));
+        public void CopyTo(TResource[] array, int arrayIndex)
+        {
+            // TODO: Improve exceptions
+            if (array == null) throw new ArgumentNullException();
+            if (arrayIndex < 0) throw new ArgumentOutOfRangeException();
+            if (Count > array.Length - arrayIndex) throw new ArgumentException();
+            for (int i = 0; i < Count; i++)
+            {
+                array[arrayIndex + i] = _stringToName(_underlyingList[i]);
+            }
+        }
+        public IEnumerator<TResource> GetEnumerator() => _underlyingList.Select(_stringToName).GetEnumerator();
+        System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() => GetEnumerator();
+        public int IndexOf(TResource value) => _underlyingList.IndexOf(_nameToString(value));
+        public void Insert(int index, TResource item) => _underlyingList.Insert(index, _nameToString(item));
+        public bool Remove(TResource item) => _underlyingList.Remove(_nameToString(item));
+        public void RemoveAt(int index) => _underlyingList.RemoveAt(index);
+    }
+
+    /// <summary>
```

Comment:
There's no qualified version here. I can't remember what we were going to do about those.